src/GafferScene/SceneAlgo.cpp

```diff
@@ -571,15 +571,15 @@ SceneProcessor *objectTweaksWalk( const SceneAlgo::History *h )
 	return nullptr;
 }
 
-ShaderTweaks *shaderTweaksWalk( const SceneAlgo::History *h, const IECore::InternedString &attributeName )
+ShaderTweaks *shaderTweaksWalk( const SceneAlgo::AttributeHistory *h )
 {
 	if( auto tweaks = h->scene->parent<ShaderTweaks>() )
 	{
 		if( h->scene == tweaks->outPlug() )
 		{
 			Context::Scope contextScope( h->context.get() );
 			if(
-				StringAlgo::matchMultiple( attributeName, tweaks->shaderPlug()->getValue() ) &&
+				StringAlgo::matchMultiple( h->attributeName, tweaks->shaderPlug()->getValue() ) &&
```

Comment:
Do I read this right, that it means we'll ignore a tweaks node that tweaks a target shuffled attribute before it was shuffled? Is this to prevent inadvertently tweaking a shuffle-copy style shuffle, whereby we may then also be tweaking an unrelated attr?

Reply:
No, the opposite. We're now using `h->attributeName`, which is the name the attribute has at this point in the history. So even if you do `ShaderTweaks->ShuffleAttributes`, we'll now find the right tweak node.

Reply:
Ahh yes, of course. I guess the second part still stands - is that dangerous? Though I guess no more than also following CopyAttributes...

Reply:
I don't think it's any more dangerous than anything else really, it's just more accurate. Even with the old behaviour, we can still be backtracking to a tweaks node at a fork in the graph, and your edits will affect multiple downstream branches while you're only viewing one. I expect the "History->Edit Tweaks..." thing will get phased out once we're making tweaks via EditScopes anyway...